### PR TITLE
- fix a false positive for endless method definition

### PR DESCRIPTION
**Pull request by koic**
Fixes: #795.

This PR fixes false positives that comparison methods recognizes as a setter method.

### CODE REVIEW

File: lib/parser/ruby30.y [hunk 1]
```diff
@@ -3071,7 +3071,7 @@ require 'parser'
   end
 
   def endless_method_name(name_t)
-    if name_t[0].end_with?('=')
+    if !%w[== === >= <= !=].include?(name_t[0]) && name_t[0].end_with?('=')
```
Comment (iliabylich) [inline lib/parser/ruby30.y:3074]: `[]=` should also be in this list (and who knows what else):
``` ruby
> ObjectSpace.each_object(Class).flat_map(&:instance_methods).uniq.select { |mid| mid.to_s.end_with?('=') && !(mid.to_s[-2] =~ /\w/) }
 => [:===, :==, :!=, :<=, :>=, :[]=]
```

File: test/test_parser.rb [hunk 2]
```diff
@@ -10107,6 +10107,73 @@ def test_endless_setter
       SINCE_3_0)
   end
 
+  def test_endless_comparison_method
+    assert_parses(
```
Comment (iliabylich) [inline test/test_parser.rb:10111]: could you rewrite it into a loop that iterates over method names please? Also, source maps are redundant here.

File: lib/parser/ruby30.y (hunk 1 above)
Comment (iliabylich) [reply lib/parser/ruby30.y:3074]: wait, or is it a setter? 😄

Comment (koic) [reply lib/parser/ruby30.y]: `[]=` will also result in the error on MRI :-)

```ruby
% ruby -ve "def []=(index, value) = 42"
ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-darwin19]
-e:1: setter method cannot be defined in an endless method definition
def []=(index, value) = 42
```

```ruby
% ruby -ve "def []=(index, value) = 42"
ruby 3.1.0dev (2021-04-06T07:03:20Z master 31ba817887) [x86_64-darwin19]
-e:1: setter method cannot be defined in an endless method definition
def []=(index, value) = 42
```

Comment (koic) [reply lib/parser/ruby30.y]: Oh, you're right. It's a setter, so it's an error and it's expected. I updated this PR again.

**Comment by iliabylich** [comment]
@koic Thanks!

**Comment by koic** [comment]
Thanks too!

**Pull request merged**